Silence config parsing warnings

When using multi-broker configs (or configs that subscribe to multiple exchanges on the same broker), we get config parsing warnings. As we add more multi-exchange and multi-broker configs, the warnings get a bit overwhelming. It would be nice to have a way to silence them.

E.g. this config:

```
# Redundant brokers
broker     amqps://USERNAME@normal-broker/
exchange   xs_USERNAME
queueName  q_${BROKER_USER}.${PROGRAM}.${CONFIG}.ddsr
subtopic   #

broker     amqps://USERNAME@backup-broker/
exchange   xs_USERNAME
queueName  q_${BROKER_USER}.${PROGRAM}.${CONFIG}.ddsr
subtopic   #
```

Causes these logs:

```
2025-11-19 20:32:46,955 63471 [WARNING] sarracenia.config parse_line ./get-stuff.conf:11 broker needs to appear before *subtopic* unless you need different queues to have different settings
2025-11-19 20:32:46,955 63471 [WARNING] sarracenia.config parse_line ./get-stuff.conf:12 exchange needs to appear before *subtopic* unless you need different queues to have different settings
2025-11-19 20:32:46,955 63471 [WARNING] sarracenia.config parse_line ./get-stuff.conf:13 queueName usually should be before subtopic in configs: subtopic to default queue
```

And the logs appear even when working with other configs.

https://github.com/MetPX/sarracenia/blob/d349a8366e99b1416ec8960d4cde226b1b81711d/sarracenia/config/__init__.py#L1611-L1620

---

I'm not sure what the best way to silence them would be. Maybe an option `silenceSubtopicWarnings`? 

Shellcheck has a syntax where you can put a comment before a line that would trigger an error/warning to disable it, for example:

```
# shellcheck disable=SC2086,SC2029
ssh ${ssh_args} ${hostname} "$@"
```

But I think that would be tricky to implement (we'd have to keep track of the previous line), and would add a bunch of extra lines:

```
# Redundant brokers
broker     amqps://USERNAME@normal-broker/
exchange   xs_USERNAME
queueName  q_${BROKER_USER}.${PROGRAM}.${CONFIG}.ddsr
subtopic   #

# disable_warning
broker     amqps://USERNAME@backup-broker/
# disable_warning
exchange   xs_USERNAME
# disable_warning
queueName  q_${BROKER_USER}.${PROGRAM}.${CONFIG}.ddsr
subtopic   #
```

--- a/sarracenia/config/__init__.py
+++ b/sarracenia/config/__init__.py
@@ -1615,6 +1615,9 @@ class Config:
                     logger.warning( f"{','.join(self.files)}:{lineno} {k} needs to be near the start of the file." )
             return
 
+        if k == 'broker':
+            self.subtopic_seen=False
+
         if k in queue_options and hasattr(self,'subtopic_seen') and self.subtopic_seen:
             logger.warning( f"{','.join(self.files)}:{lineno} {k} needs to appear before *subtopic*" \
                 " unless you need different queues to have different settings")

